odl/tomo/geometry/detector.py

```diff
@@ -51,38 +50,27 @@ class Detector(with_metaclass(ABCMeta, object)):
     * optionally a sampling grid for the parameters
     """
 
-    def __init__(self, ndim, params, grid=None):
+    def __init__(self, ndim, grid):
         """Initialize a new instance.
 
         Parameters
         ----------
         ndim : non-negative `int`
             The number of dimensions of the detector
-        params : `IntervalProd`
-            The parameter set defining the detector area
         grid : `TensorGrid`, optional
-            A sampling grid for the parameter set, in which it must be
-            contained
+            A sampling grid for the parameter set (pixels)
         """
-        if not isinstance(params, IntervalProd):
-            raise TypeError('parameter set {!r} is not a an interval product.'
-                            ''.format(params))
 
-        if params.ndim != ndim:
-            raise ValueError('parameters {!r} are not {}-dimensional.'
-                             ''.format(params, ndim))
+        if not isinstance(grid, TensorGrid):
+            raise TypeError('grid {!r} is not a `TensorGrid` instance.'
+                            ''.format(grid))
 
-        if grid is not None:
-            if not isinstance(grid, TensorGrid):
-                raise TypeError('grid {!r} is not a `TensorGrid` instance.'
-                                ''.format(grid))
-
-            if not params.contains_set(grid):
-                raise ValueError('grid {!r} not contained in parameter set '
-                                 '{!r}.'.format(grid, params))
+        if grid.ndim != ndim:
+            raise ValueError('grid {!r} are not {}-dimensional.'
```

Comment:
Simply use  `'{} is not {}-dimensional.'` The grid pretty-prints itself with "grid"